lib/git_worktree.rb

```diff
@@ -201,17 +205,47 @@ def checkout(target_directory)
     @repo.checkout_tree(tree, :target_directory => target_directory, :strategy => :force)
   end
 
-  def credentials_cb(url, _username, _types)
-    if @credentials_set
-      raise GitWorktreeException::InvalidCredentials, "Please provide username and password for URL #{url}" if @username.blank? || @password.blank?
-      raise GitWorktreeException::InvalidCredentials, "Invalid credentials for URL #{url}"
+  def with_credential_options
+    if @ssh_private_key
+      @ssh_private_key_file = Tempfile.new
+      @ssh_private_key_file.write(@ssh_private_key)
+      @ssh_private_key_file.close
+    end
+
+    options = {:credentials => method(:credentials_cb)}
+    options[:certificate_check] = @certificate_check_cb if @certificate_check_cb
+
+    yield options
+  ensure
+    if @ssh_private_key_file
+      @ssh_private_key_file.unlink
```

Comment:
I don't think I have a better way of doing this, but effectively for every action that requires this method, we are going to be creating and deleting the `@ssh_private_key_file`, correct?  Might be a bit of extra trashing of the file system as a result.

Reply:
Yeah, but only for remote interactions, so it only happens once per invocation.  I thought the same as you that it might thrash but I think we can deal with that later if it's really an issue.

Reply:
Yeah, kinda assumed.  I am not going to "resolve this" as I do want it more visible for others in the future, but I think it can stay as is until it becomes a problem.